backend/src/database/migrations/20221012051800-add-api-key-hash-column-to-user.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.addColumn('users', 'api_key_hash', {
+      type: Sequelize.DataTypes.STRING(255),
+      allowNull: true,
+    })
+    await queryInterface.sequelize.query(
+      'UPDATE users SET api_key_hash = api_key;'
```

Comment:
Trigger should be created before backfill, otherwise there's still a slim chance a new insert on users table happened before the trigger has been created resulting in a row without `api_key_hash`

Reply:
Hmm I see your point, but I don't think it's relevant here. 

This new trigger (`format_user_email`) is meant to replace `format_user_on_insert`, which (1) normalizes user emails by ensuring they're lower case; and (2) if the manual edit to the user row did not specify `api_key`, i.e. it is `''`, it'll be automatically set to `NULL`.

What `format_user_email` does is to only do (1) and not do (2). I don't think (2) is necessary and was in fact a hidden coupling that was a hindrance to this refactoring (I ran into an error and had to debug to find out that `api_key` was referred to in this db function.

When this trigger is created, both `format_user_email` and `format_user_on_insert` would exist concurrently (the latter is only removed in part 2), so there is no risk of any missed insertion.

Hope this makes sense?

Reply:
Thanks for the clarification!!